sample-app/src/main/java/com/splunk/android/sample/FirstFragment.java

```diff
@@ -84,11 +85,20 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         });
 
         binding.httpMe.setOnClickListener(v -> {
-            try (WorkflowTimer workflow = splunkRum.startWorkflow("Custom Workflow")) {
-                backgrounder.submit(() -> makeCall("https://ssidhu.o11ystore.com/"));
-            }
+            Span workflow = splunkRum.startWorkflow("Custom Workflow");
+            backgrounder.submit(() -> {
+                makeCall("https://ssidhu.o11ystore.com/");
+                workflow.end();
+            });
+        });
+        binding.httpMeBad.setOnClickListener(v -> {
+            Span workflow = splunkRum.startWorkflow("Workflow With Error");
+            backgrounder.submit(() -> {
+                makeCall("https://asdlfkjasd.asdfkjasdf.ifi");
```

Comment:
If the `makeCall` throws an exception, we will never make it to the `workflow.end()` below -- is that ok, or would it potentially "leak" a span? Do blocks like this need to be done in try...finally?

Reply:
generally, yes. Right now the `makeCall` catches anything that might be thrown, so it's probably ok for the sample app?  I agree it's not ideal, and this whole thing needs a bit of a rework to really be considered "good code".

There isn't really any span "leakage" in otel unless you hold a reference to it, since nothing in the SDK keeps a reference itself.